modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/configuration/ScimClientOAuth2ApplicationConfigurationFactory.java

```diff
@@ -111,11 +112,10 @@ protected void deactivate(Integer reason) throws PortalException {
 	private OAuth2Application _getOrAddOAuth2Application(
 			long companyId,
 			ScimClientOAuth2ApplicationConfiguration
-				scimClientOAuth2ApplicationConfiguration)
+				scimClientOAuth2ApplicationConfiguration,
+			long userId)
 		throws Exception {
 
-		User user = _userLocalService.getGuestUser(companyId);
-
 		User clientCredentialUser = _userLocalService.getUserByScreenName(
```

Comment:
Do we also need to change this `clientCredentialUser` to the current user?  Could you please try adjusting this and see if everything works as intended?